internal/containerd/install.go

```diff
@@ -34,20 +29,19 @@ type Source interface {
 	GetContainerd(version string) artifact.Package
 }
 
-func Install(ctx context.Context, tracker *tracker.Tracker, source Source, containerdSource SourceName) error {
-	if containerdSource == ContainerdSourceNone {
+func Install(ctx context.Context, artifactsTracker *tracker.Tracker, source Source, containerdSource tracker.SourceName) error {
+	if containerdSource == tracker.ContainerdSourceNone || areContainerdAndRuncInstalled() {
```

Comment:
This is where i got lost the other day. help me understand in what case the condition for `areContainerdAndRuncInstalled` is useful: why would we skip installation if the user is asking as to install but they have something already installed? In my head we would just error if the user already has it but they are not using option "none"


or is this for the upgrade flow where we need to deal with a tracker file with an empty `containerdSource`?